cmd/uplink/cmd/setup.go

```diff
@@ -251,3 +260,37 @@ func ApplyDefaultHostAndPortToAddr(address, defaultAddress string) (string, erro
 	// address is host:
 	return net.JoinHostPort(addressParts[0], defaultPort), nil
 }
+
+// SaveEncryptionKey saves the key in a new file which will be stored in
+// filepath.
+// It returns an error if the directory doesn't exist, the file already exists
+// or there is an I/O error.
+func SaveEncryptionKey(key []byte, filepath string) (err error) {
+	f, err := os.OpenFile(filepath, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0600)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return errors.New("directory path doesn't exist")
+		}
+
+		if os.IsExist(err) {
+			return errors.New("file key already exists")
+		}
+
+		return err
+	}
+
+	defer func() {
```

Comment:
`defer func () { err = errs.Combine(err, f.Close()) }()`

Reply:
is it one line readable enough?

Reply:
It's not really readable, however it's a very common thing we do... so we can easily recognize this part by learning it once. You can split into multiple lines as well.

Reply:
Bill also prefers fewer lines.  😄

Reply:
This kind of things are the bad things that could make any convention easy to remember and the harder are to remember the harder is to have contributions aligned with them.